### PR TITLE
fix:ListObjects and ListObjectV2 correctly handles unordered and delimiter

### PR DESCRIPTION
**Pull request by LeonWang0735**
## Type of Change
- [ ] New Feature
- [x] Bug Fix
- [ ] Documentation
- [ ] Performance Improvement
- [ ] Test/CI
- [ ] Refactor
- [ ] Other:

## Summary of Changes

Fix S3 compatibility validation: properly reject the allow-unordered=true query parameter when used with Delimiter="/" in ListObjectsV2 operations, returning InvalidArgument error as required by the S3 API specification to fix the failing test_bucket_list_unordered and test_bucket_listv2_unorderedtest case.

## Checklist
- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines
- [ ] Passed `make pre-commit`
- [ ] Added/updated necessary tests
- [ ] Documentation updated (if needed)
- [ ] CI/CD passed (if applicable)

## Impact
- [ ] Breaking change (compatibility)
- [ ] Requires doc/config/deployment update
- [x] Other impact:Improves S3 API compliance by enforcing parameter validation; expected to be backward-compatible for compliant clients.

### CODE REVIEW

File: rustfs/src/storage/ecfs.rs [hunk 1]
```diff
@@ -2816,6 +2816,29 @@ impl S3 for FS {
...
+                        let decoded_value = urlencoding::decode(value).unwrap_or_else(|_| value.into());
+                        decoded_key.trim() == "allow-unordered"
+                            && decoded_value.trim() == "true"
+                    } else {
+                        false
+                    }
+                });
+
+                if has_allow_unordered {
+                    println!("has_allow_unordered");
```
Comment (houseme) [inline rustfs/src/storage/ecfs.rs]: Log records do not use println processing, thank you!

**Comment by houseme** [comment]
This PR correctly fixes an S3 API compatibility issue by rejecting the invalid `allow-unordered=true` parameter when `delimiter="/"` is specified, aligning with AWS S3 specifications and fixing failing tests. The change is small, targeted, and improves compliance. However, minor issues need addressing for production readiness.

#### Key Issues and Suggestions
- **Remove debug code**: The `println!("has_allow_unordered");` statement must be removed as it is inappropriate for production code. Replace with proper logging if debugging is needed.
- **Improve parameter parsing**: The manual query string parsing (using `split` and `split_once`) is error-prone and could fail on malformed inputs or URL encoding issues. Consider using a URL parsing crate (e.g., `url`) or the framework's built-in query parameter handling for robustness.
- **Enhance error message**: Update the error message to be more descriptive, e.g., "InvalidArgument: The allow-unordered parameter cannot be used when delimiter is specified." Include a reference to the S3 spec for clarity.
- **Add tests**: While existing tests are fixed, add unit tests for edge cases (e.g., `allow-unordered=false`, empty delimiter, malformed queries) to ensure comprehensive coverage. Update the PR checklist to reflect this.
- **Code documentation**: Add comments explaining the validation logic and why it's required per S3 specs.

#### Recommendations
- **Request Changes**: Address the debug code removal and parameter parsing improvements before merging to ensure code quality and maintainability.
- **Optional Enhancements**: Extract the validation into a reusable function and verify performance impact in high-load scenarios.
- **Approval Status**: Pending fixes. Re-review after updates.

Thanks for the contribution! This strengthens S3 compatibility. Let me know if you need help implementing these changes.

**Comment by LeonWang0735** [comment]
Related to #1289
